.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -687,14 +687,17 @@ private DateTimeResolutionResult ParseSingleTimePoint(string text, DateObject re
                     ret.Success = true;
                 }
 
-                if (dateContext != null && text != "today")
+                // check if text includes special day
+                var specialDayMatch = this.config.SpecialDayRegex.Match(text);
+
+                if (dateContext != null && !specialDayMatch.Success)
```

Comment:
How often is dateContext null? Why not check this first, as it could potentially avoid running the SpecialDayRegex unnecessarily?

Reply:
I need to check it and then give a reply.